nvdaHelper/remote/inProcess.cpp

```diff
@@ -156,18 +157,32 @@ void CALLBACK inProcess_winEventCallback(HWINEVENTHOOK hookID, DWORD eventID, HW
 	}
 }
 
-void execInWindow(HWND hwnd, execInWindow_funcType func,void* data) {
-	// Using SendMessage here causes outgoing cross-process COM calls to fail with RPC_E_CANTCALLOUT_ININPUTSYNCCALL,
-	// which breaks us for Firefox multi-process. See Mozilla bug 1297549 comments 14 and 18.
-	// Use SendMessageCallback instead.
-	SENDASYNCPROC callback = [] (HWND hwnd, UINT msg, ULONG_PTR data, LRESULT result) {
-		// Signal the waiting message loop to exit.
-		PostQuitMessage(0);
-	};
-	if (!SendMessageCallback(hwnd,wm_execInWindow,(WPARAM)&func,(LPARAM)data, callback, NULL))
-		return;
-	MSG msg;
-	// Wait in a message loop until the callback fires and sends WM_QUIT.
-	// We also abort the loop if WaitMessage fails for some reason.
-	while (GetMessage(&msg, NULL, WM_QUIT, WM_QUIT) > 0);
+bool execInThread(long threadID, execInThread_funcType func) {
+	HANDLE handles[2]={NULL,NULL};
+	handles[0]=CreateEvent(NULL,TRUE,FALSE,NULL);
+	if(!handles[0]) {
+		LOG_DEBUGWARNING(L"Could not create event");
+		return false;
+	}
+	handles[1]=OpenThread(SYNCHRONIZE,false,threadID);
+	if(!handles[1]) {
+		LOG_DEBUGWARNING(L"Could not open UI thread");
+		CloseHandle(handles[0]);
+		return false;
+	}
+	if(PostThreadMessage(threadID,wm_execInThread,(WPARAM)&func,(LPARAM)(handles[0]))==0) {
+		LOG_DEBUGWARNING(L"Failed to post thread message");
+		CloseHandle(handles[0]);
+		CloseHandle(handles[1]);
+		return false;
+	}
+	if(WaitForMultipleObjects(2,handles,false,INFINITE)!=WAIT_OBJECT_0) {

```

Comment:
Would it be useful to know specifically that the thread died while waiting (WAIT_OBJECT_0 + 1) as opposed to some other error? Not sure, but thought I'd raise it as food for thought.